internal/db/stats_test.go

```diff
@@ -45,70 +46,210 @@ func (this *DBTests) DBTestStoreSystemStats(test *testing.T) {
+			stats.User_Email_Key:    "U",
+			stats.Assignment_ID_Key: "A",
+			stats.Course_ID_Key:     "C",
+			stats.Locator_Key:       "11",
 		},
-		Type:         stats.CourseMetricTypeGradingTime,
-		CourseID:     "C",
-		AssignmentID: "A",
-		UserEmail:    "U",
-		Value:        100,
 	}
 
-	query := stats.CourseMetricQuery{
-		CourseID: "C",
+	query := stats.Query{
+		Type: stats.API_Request_Stats_Type,
 	}
 
-	err := StoreCourseMetric(&testRecord)
-	if err != nil {
-		test.Fatalf("Failed to store stats: '%v'.", err)
+	runStoreStatsTests(test, &metric, query)
+}
+
+func (this *DBTests) DBTestStoreTaskTimeStats(test *testing.T) {
+	Clear()
+	defer Clear()
+
+	metric := stats.Metric{
+		Timestamp: timestamp.Now(),
+		Type:      stats.Task_Time_Stats_Type,
+		Value:     float64(100),
+		Attributes: map[stats.MetricAttribute]any{
+			stats.User_Email_Key:    "U",
+			stats.Assignment_ID_Key: "A",
+			stats.Course_ID_Key:     "C",
+		},
 	}
 
-	records, err := GetCourseMetrics(query)
-	if err != nil {
-		test.Fatalf("Failed to fetch stats: '%v'.", err)
+	query := stats.Query{
+		Type: stats.Task_Time_Stats_Type,
 	}
 
-	if len(records) != 1 {
-		test.Fatalf("Did not get the correct number of records. Expected: 1, Actual: %d.", len(records))
+	runStoreStatsTests(test, &metric, query)
+}
+
+func (this *DBTests) DBTestStoreGradingTimeStats(test *testing.T) {
+	Clear()
+	defer Clear()
+
+	metric := stats.Metric{
+		Timestamp: timestamp.Now(),
+		Type:      stats.Grading_Time_Stats_Type,
+		Value:     float64(100),
+		Attributes: map[stats.MetricAttribute]any{
+			stats.User_Email_Key:    "U",
+			stats.Assignment_ID_Key: "A",
+			stats.Course_ID_Key:     "C",
+		},
 	}
 
-	if !reflect.DeepEqual(*records[0], testRecord) {
-		test.Fatalf("Did not get the expected record back. Expected: '%s', Actual: '%s'.",
-			util.MustToJSONIndent(testRecord), util.MustToJSONIndent(*records[0]))
+	query := stats.Query{
+		Type: stats.Grading_Time_Stats_Type,
 	}
+
+	runStoreStatsTests(test, &metric, query)
 }
 
-func (this *DBTests) DBTestStoreAPIRequestMetrics(test *testing.T) {
+func (this *DBTests) DBTestStoreCodeAnalysisTimeStats(test *testing.T) {
 	Clear()
 	defer Clear()
 
-	testRecord := stats.APIRequestMetric{
-		BaseMetric: stats.BaseMetric{
-			Timestamp: timestamp.Now(),
+	metric := stats.Metric{
+		Timestamp: timestamp.Now(),
+		Type:      stats.Code_Analysis_Time_Stats_Type,
+		Value:     float64(100),
+		Attributes: map[stats.MetricAttribute]any{
+			stats.User_Email_Key:    "U",
+			stats.Assignment_ID_Key: "A",
+			stats.Course_ID_Key:     "C",
 		},
-		Sender:       "2",
-		Endpoint:     "E",
-		UserEmail:    "U",
-		CourseID:     "C",
-		AssignmentID: "A",
-		Locator:      "1",
-		Duration:     100,
 	}
 
-	err := StoreAPIRequestMetric(&testRecord)
+	query := stats.Query{
+		Type: stats.Code_Analysis_Time_Stats_Type,
+	}
+
+	runStoreStatsTests(test, &metric, query)
+}
+
+func (this *DBTests) DBTestStoreMetricFailures(test *testing.T) {
```

Comment:
These tests are strange.

The first one is not a "failure" as the test name implies.
It's just a non-matching query.

Then since you are focused on the storing, I don't think you should specify the query in the test.
You can just make a query that matches the type of the metric you are storing.